pkg/eval/evaltest/evaltest.go

```diff
@@ -122,9 +123,16 @@ func (c Case) Throws(reason error, stacks ...string) Case {
 }
 
 // DoesNotCompile returns an altered Case that requires the source code to fail
-// compilation.
-func (c Case) DoesNotCompile() Case {
-	c.want.CompilationError = anyError{}
+// compilation with a specific error. It accepts a string or a MatchingRegexp{}.
+func (c Case) DoesNotCompile(ce any) Case {
+	switch v := ce.(type) {
+	case string:
+		c.want.CompilationError = stringError{isRegexp: false, str: v}
+	case MatchingRegexp:
+		c.want.CompilationError = stringError{isRegexp: true, str: v.Pattern}
```

Comment:
Regular expressions are used in exactly one test case. Best to just stick to matching the full message.